### PR TITLE
feat(api): new endpoint PartialUpdateService

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New endpoint PartialUpdateService:
```
PATCH /configuration/services/{service_id}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by arnapou — APPROVED** [review]

File: centreon/src/Core/Service/Domain/Model/Service.php [hunk 1]
```diff
@@ -544,4 +544,440 @@ public function getGeoCoords(): ?GeoCoords
     {
         return $this->geoCoords;
     }
+
+    public function isNameIdentical(string $testName): bool {
```
[the remark this answers is not in the document]
Comment (a-launois) [reply centreon/src/Core/Service/Domain/Model/Service.php:548]: I used the same wording as what can be found in PartialUpdateHost & PartialUpdateHostTemplate

[the remark this answers is not in the document]
Comment (arnapou) [reply centreon/src/Core/Service/Domain/Model/Service.php:548]: Perfect. TY for the clarification.

**Review by arnapou — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

File: centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        $this->timePeriodRepository = $this->createMock(ReadTimePeriodRepositoryInterface::class),
+        $this->imageRepository = $this->createMock(ReadViewImgRepositoryInterface::class),
+        $this->readHostRepository = $this->createMock(ReadHostRepositoryInterface::class),
+        $this->readServiceCategoryRepository = $this->createMock(ReadServiceCategoryRepositoryInterface::class),
+        $this->readServiceGroupRepository = $this->createMock(ReadServiceGroupRepositoryInterface::class),
+        $this->contact = $this->createMock(ContactInterface::class),
+    );
+});
+
+it('should raise an exception when the new name already exist', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php:59]:
```suggestion
it('should raise an exception when the new name already exists', function (): void {
```

File: centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        ->method('exist')
+        ->willReturn([2]);
+
+    $this->validation->assertAreValidGroups($serviceGroupIds);
+})->throws(
+    ServiceException::class,
+    ServiceException::idsDoNotExist('service_groups', [1, 3])->getMessage()
+);
+
+it('should raise an exception when the service groups IDs do not exist, as a non-administrator', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php:255]:
```suggestion
it('should raise an exception when the service group IDs do not exist, as a non-administrator', function (): void {
```

File: centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        ->method('findAllExistingIdsByAccessGroups')
+        ->willReturn([2]);
+
+    $this->validation->assertAreValidCategories([1, 2]);
+})->throws(
+    ServiceException::class,
+    ServiceException::idsDoNotExist('service_categories', [1])->getMessage()
+);
+
+it('should raise an exception when the service groups IDs do not exist, as an administrator', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Service/Application/UseCase/PartialUpdateService/PartialUpdateServiceValidationTest.php:236]:
```suggestion
it('should raise an exception when the service group IDs do not exist, as an administrator', function (): void {
```

**Review by arnapou — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**